var/spack/repos/builtin/packages/fenics2019/package.py

```diff
@@ -0,0 +1,144 @@
+    variant('hdf5',         default=True,  description='Compile with HDF5')
+    variant('parmetis',     default=True,  description='Compile with ParMETIS')
+    variant('scotch',       default=True,  description='Compile with Scotch')
+    variant('petsc',        default=True,  description='Compile with PETSc')
+    variant('slepc',        default=True,  description='Compile with SLEPc')
+    variant('trilinos',     default=False,  description='Compile with Trilinos')
+    variant('suite-sparse', default=True,
+            description='Compile with SuiteSparse solvers')
+    variant('vtk',          default=False, description='Compile with VTK')
+    variant('qt',           default=False, description='Compile with QT')
+    variant('mpi',          default=True,
+            description='Enables the distributed memory support')
+    variant('openmp',       default=True,
+            description='Enables the shared memory support')
+    variant('shared',       default=True,
+            description='Enables the build of shared libraries')
+    variant('doc',          default=False,
+            description='Builds the documentation')
+    variant('build_type', default='RelWithDebInfo',
+            description='The build type to build',
+            values=('Debug', 'Release', 'RelWithDebInfo',
+                    'MinSizeRel', 'Developer'))
+
+    # apply patch to fix header issue with latest boost library
+    patch('header_fix.patch')
+
+    extends('python', when='+python')
+
+    depends_on('python@3.7.8:', type=('build', 'run'), when='+python')
+
+    # build python components of FEniCS
+    depends_on('py-fenics-ffc', type=('build'), when='~python')
+    depends_on('py-fenics-ffc', type=('build', 'run'), when='+python')
+    depends_on('py-fenics-dijitso', type=('build', 'run'), when='+python')
+
+    depends_on('eigen@3.3.7:')
+    depends_on('pkg-config')
+    depends_on('zlib')
+    depends_on('boost+filesystem+program_options+system+iostreams+timer+regex+chrono')
+
+    depends_on('mpi', when='+mpi')
+    depends_on('hdf5+hl+fortran', when='+hdf5+petsc')
+    depends_on('hdf5+hl', when='+hdf5~petsc')
+    depends_on('metis+real64', when='+parmetis')
+    depends_on('parmetis', when='+parmetis')
+    depends_on('scotch~metis', when='+scotch~mpi')
+    depends_on('scotch+mpi~metis', when='+scotch+mpi')
+    depends_on('petsc', when='+petsc')
+    depends_on('slepc', when='+slepc')
+    depends_on('py-petsc4py@3.6:', when='+petsc+python')
+    depends_on('trilinos', when='+trilinos')
+    depends_on('vtk', when='+vtk')
+    depends_on('suite-sparse@4.5.6', when='+suite-sparse')
+    depends_on('qt', when='+qt')
+
+    depends_on('py-pybind11', type=('build', 'run'))
+    depends_on('cmake@3.17.3:', type='build')
+
+    depends_on('py-setuptools', type='build', when='+python')
+    depends_on('py-pkgconfig', type='run', when='+python')
+    depends_on('py-sphinx@1.0.1:', when='+doc', type='build')
+
+    def cmake_is_on(self, option):
```

Comment:
Could you replace this with the builtin `self.define_from_variant`?

https://spack.readthedocs.io/en/latest/build_systems/cmakepackage.html#adding-flags-to-cmake

Reply:
agreed, will be changed in next commit